add the machine learning blogpost (#13)
hashintel/hash: resources/glossary/machine-learning.mdx

---
title: Machine Learning
description: general introduction to how Machine Learning could be applied in the context of multi-agent simulations
slug: machine-learning
---

## WHAT IS MACHINE LEARNING

> **Machine Learning** is a sub-field of Artificial Intelligence where algorithms learn from interacting with an **environment** or from a **training dataset**. Machine Learning is primarily used to make **predictions** using statistics, applying those statitical methods to business problems is the realm of **predictive analytics**. Most recent advances in Machine Learning have been driven by research in Deep Learning, which is the combination of Machine Learning and multi-layer [artificial neural networks](https://hash.ai/wiki/neural-nets).

Reviewer: ```suggestion
**Machine Learning** is a sub-field of Artificial Intelligence where algorithms learn from interacting with an **environment** or from a **training dataset**. Machine Learning is primarily used to make **predictions** using statistics, applying those statistical methods to business problems is the realm of **predictive analytics**. Most recent advances in Machine Learning have been driven by research in Deep Learning, which is the combination of Machine Learning and multi-layer [artificial neural networks](https://hash.ai/glossary/neural-nets).
```

<p align="center">

> ![Artificial Intelligence Machine Learning Deep Learning](https://i.imgur.com/xatMpJa.png)

Reviewer: 1. We should move images into the GitHub repo for this PR to merge.
2. I've captured a followup card to redesign this illustration in HASH house style, but this second bullet shouldn't block merge.

Author: Makes sense. I haven't found any place to insert images for now. What do you mean by HASH house style? Using the same palette as hash.ai's logo / landing (blue/pink)?

Reviewer: Re: best place to upload resource-site imagery, I've opened an internal thread under #simulations w/ Ciaran.

By style I mean in keeping with https://cdn-us1.hash.ai/site/hash-des-flow.svg (from https://hash.ai/glossary/discrete-event-modeling). There are some other figures within the glossary that also need redoing (e.g. https://hash.ai/glossary/system-dynamics) so we shouldn't consider the mismatch a blocker. I've captured the followup, so no action required on your part here.

</p>

## WHY USE MACHINE LEARNING

> The advantage of using Machine Learning instead of traditional Artificial Intelligence algorithms is that models can act and adapt to different possible environments or data without being explicitly programmed to do so. In that sense, it is more flexible than rule-based Artificial Intelligence where humans often have to hard-code expert knowledge about the task at hand. A striking example has been AlphaZero beating the best (overly engineered) chess programs without any prior rule or knowledge about chess.

Author: Makes sense, added this to the marketing onboarding task after this!

The two use-cases of machine learning are to either give a numeric answer to a question (eg. "what will be the price of bitcoin tomorrow"), which we call a **regression problem**, the simpler example being a linear regression (aka a straight line passing through the data). The second use-case is **classification** where the objective is to tell if a given input belongs to a particular group. A popular example is to determine whether in an image there is a cat or a dog.

## TYPES OF MACHINE LEARNING

> The three most important approaches in machine learning are supervised learning, unsupervised learning and reinforcement learning. Other methods such as [data mining](https://hash.ai/wiki/data-mining), semi-supervised learning or feature learning make use of those three essential methods listed above. Although Deep Learning has been the most popular paradigm in Machine Learning in the past 5 years, traditionally machine learning has made extensive use of decision trees, support-vector machines, bayesian networks and genetic algorithms. 

Reviewer: We should avoid making positionally relative references to content/information on-page (e.g. in this case saying 'listed above').

It's unlikely content is printed out/even more unlucky if then split (and thus 'beneath' would become 'overleaf'), but we can avoid the possibility entirely by rephrasing things. @mtrazzi Please have a stab at this

Reviewer: To make the second sentence less scary/word-soupy we might also add "(defined below)" to the end of it.

> The three most important approaches in machine learning are supervised learning, unsupervised learning and reinforcement learning. Other methods such as [data mining](https://hash.ai/wiki/data-mining), semi-supervised learning or feature learning make use of those three essential methods listed above. Although Deep Learning has been the most popular paradigm in Machine Learning in the past 5 years, traditionally machine learning has made extensive use of decision trees, support-vector machines, bayesian networks and genetic algorithms. 

Reviewer: ```suggestion
The three most important approaches in machine learning are supervised learning, unsupervised learning and reinforcement learning. Other methods such as [data mining](https://hash.ai/glossary/data-mining), semi-supervised learning or feature learning make use of those three essential methods listed above. Although Deep Learning has been the most popular paradigm in Machine Learning in the past 5 years, traditionally machine learning has made extensive use of decision trees, support-vector machines, bayesian networks and genetic algorithms. 
```

1. **Supervised Learning**: the model learns from a training data with labels in order to be able to predict the labels of some test data, where the algorithm has not seen the labels. This can be for instance a classifier being able to tell if an image is a dog or a cat after having been exposed to millions of images of cats and dogs, with the corresponding labels. _Supervised_ comes from the model learning from the correct questions (training examples) and answers (training labels), like if it was a teacher supervising the learning by providing some material.

2. **Unsupervised Learning**: the model does not have access to any label. One popular application is clustering where the goal is to identify natural groups of examples in the dataset. This could be for instance determining the behavior of a website user. Another idea is to see which datapoints in our dataset is very far away from what we would expect from the rest of the distribution, which we call an **outlier**, or **anomaly** in anomaly detection. This is used in computer security to detect strange behaviors, for instance with non human-like requests.

> 3. **Reinforcement Learning**: an agent interacts with an environment by observing states and rewards from his actions. It shares some components of [agent-based modeling](https://hash.ai/wiki/agent-based-modeling) like having agents that interact with environments, and the state is essentially the collection of all the agents' properties. However, in reinforcement learning with one agent we do not consider the rules as something separate from the environment, but solely as a particular part of the environment that an agent needs to learn. In Multi-Agent Reinforcement Learning we can either consider each agent as being in its own one-agent reinforcement learning environment, or provide ways of communication between agents that would allow them to share their plans, states and rewards. For more details about reinforcement learning and how to apply it with neural networks see the glossary page on [deep reinforcement learning](https://hash.ai/wiki/deep-reinforcement-learning).

Reviewer: ```suggestion
3. **Reinforcement Learning**: an agent interacts with an environment by observing states and rewards from his actions. It shares some components of [agent-based modeling](https://hash.ai/glossary/agent-based-modeling) like having agents that interact with environments, and the state is essentially the collection of all the agents' properties. However, in reinforcement learning with one agent we do not consider the rules as something separate from the environment, but solely as a particular part of the environment that an agent needs to learn. In Multi-Agent Reinforcement Learning we can either consider each agent as being in its own one-agent reinforcement learning environment, or provide ways of communication between agents that would allow them to share their plans, states and rewards. For more details about reinforcement learning and how to apply it with neural networks see the glossary page on [deep reinforcement learning](https://hash.ai/glossary/deep-reinforcement-learning).
```

> 4. **Data Mining**: methods used in data mining broadly overlap with machine learning techniques, especially unsupervised learning. However, the main goal is to discover underlying **properties** in the training data, without any requirement of "learning". These deduced properties will then be used by predictive algorithms.

Reviewer: ```suggestion
4. **[Data Mining])(https://hash.ai/glossary/data-mining)**: methods used in data mining broadly overlap with machine learning techniques, especially unsupervised learning. However, the main goal is to discover underlying **properties** in the training data, without any requirement of "learning". These deduced properties will then be used by predictive algorithms.
```
